ksqldb-execution/src/main/java/io/confluent/ksql/execution/codegen/SqlToJavaVisitor.java

```diff
@@ -765,10 +766,11 @@ private String visitBooleanComparisonExpression(final ComparisonExpression.Type
           : "null";
 
       final String codeString = "((" + resultSchemaString + ")"
-          + functionClassName + ".searchedCaseFunction(ImmutableList.of( "
-          + StringUtils.join(lazyWhenClause, ", ") + "),"
+          + functionClassName + ".searchedCaseFunction(ImmutableList.copyOf(Arrays.asList( "
```

Comment:
I believe every time this is evaluated for a given row, it creates this list and iterates through.  It would be good to avoid additional copies if possible.  I read through the documentation for Arrays.asList and it actually seems to just wrap the array and not copy it, so I think this does just one copy, which is what the original code did.  You might want to verify that, but I think you're good.

I would make sure to add comments here documenting why this combination is necessary!  Otherwise someone will simplify it again and rebreak it.

Consider making a separate static method in `CodeGenUtil` that's documented well so that it's use stands out.  If all variadic args works fine, it could be something like:
```
public static <T> List<T> listOf(final T... a) {
    return ImmutableList.copyOf(a);
  }
```

Reply:
Let's just play it safe keep the composition of the 2 standard methods